src/ir/properties.h

```diff
@@ -51,8 +51,26 @@ inline bool isSymmetric(Binary* binary) {
     case XorInt64:
     case EqInt64:
     case NeInt64:
+
+    case EqFloat32:
+    case NeFloat32:
+    case EqFloat64:
+    case NeFloat64:
       return true;
 
+    case AddFloat32:
+    case MulFloat32:
+    case AddFloat64:
+    case MulFloat64: {
+      // We should be more careful with different forrm of NaNs so
+      // check if LHS is non-NaN constant. We are not interested in RHS
+      // in term of canonization which prefer constant on right side.
+      if (auto* c = binary->left->dynCast<Const>()) {
+        return !c->value.isNaN();
+      }
+      return false;
```

Comment:
This behavior is very specific to the use of this function for canonicalization in OptimizeInstructions, so it doesn't make sense to have in this function as long as this function is in this general utility header.

One solution would be to move this whole function out of properties.h into OptimizeInstructions, but a better solution would be to just move this special logic for floats into a new `isSymmetric` function in OptimizeInstructions that calls out to this general one for everything except the float special cases.

Reply:
Make sense!